homework_7/functional.py

```diff
@@ -0,0 +1,72 @@
+import sys
+
+def sequential_map(*args):
+    """Takes several functions and a container and returns a list 
+    of all container values after sequentional applying of functions to them"""
+    funcs = args[:-1]
+    container = args[-1]
+    for func in funcs:
+        container = list(map(func, container))
+    return container
+
+
+def consensus_filter(*args):
+    """Takes any number of functions that return True or False and a container with values
+    and returns a list of values that passed all functions with True"""
+    funcs = args[:-1]
+    container = args[-1]
+    for func in funcs:
+        container = list(filter(func, container))
+    return container
+
+
+def conditional_reduce(func1, func2, container):
+    """Takes 2 functions and a container and returns a single value 
+    which is the result of reduce after skipping values that did
+    not pass the condition of the first function"""
+    container = list(filter(func1, container))
+    result = container[0]
+    for nxt in container[1:]:
+        result = func2(result, nxt)
+    return result
+
+
+def func_chain(*args):
+    """Takes any number of functions as arguments and returns
+    a function concatenating of them by sequential execution"""
+    def result_func(val):
+        result = val
+        for func in args:
+            result = func(result)
+        return result
```

Comment:
Первая строка не нужна
```suggestion
        for func in args:
            val = func(val)
        return val
```